Map/MapManager.gd

```diff
@@ -6,14 +6,19 @@ static func create_map(): ## Generates and Populates a map with rooms that have
 	var _grid: Array = [] ## 2d array to return. this will be populated with rooms
 	
 	for index_height: int in range(_map.floors_width.size()): ## loop through elements of the grid and assign a room type
+		var _max_floor_size = _map.floors_width[_map.floors_width.size()/2]
 		_grid.append([])
+		var _padding = (_max_floor_size - _map.floors_width[index_height])/2
+		var _floor_width = _map.floors_width[index_height]
+		for index_padding:int in _padding*2:
```

Comment:
why use `_padding*2` ?
I think a much simpler way to add the padding is to Array constructors directly:
```GDScript
var _padding = Array(_padding_size)
_padding.fill(null)
_grid.append(_padding)
```
Then you add the rooms, and you add the padding again (no need to calculate it a second time, you already have it)